modules/telestion-application/src/main/java/de/wuespace/telestion/application/loader/TelestionLoader.java

```diff
@@ -0,0 +1,323 @@
+	}
+
+	@Override
+	public void onAfterVertxStartup(Promise<Void> startPromise) throws Exception {
+		onAfterVertxStartup();
+		startPromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onAfterVertxStartup(Promise)}.
+	 */
+	public void onAfterVertxStartup() throws Exception {
+	}
+
+	@Override
+	public void onBeforeVertxShutdown(Promise<Void> stopPromise) throws Exception {
+		onBeforeVertxShutdown();
+		stopPromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onBeforeVertxShutdown(Promise)}.
+	 */
+	public void onBeforeVertxShutdown() throws Exception {
+	}
+
+	@Override
+	public void onAfterVertxShutdown(Promise<Void> stopPromise) throws Exception {
+		onAfterVertxShutdown();
+		stopPromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onAfterVertxShutdown(Promise)}.
+	 */
+	public void onAfterVertxShutdown() throws Exception {
+	}
+
+	@Override
+	public void onExit(Promise<Void> stopPromise) throws Exception {
+		onExit();
+		stopPromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onExit(Promise)}.
+	 */
+	public void onExit() throws Exception {
+	}
+
+	@Override
+	public void onBeforeVerticleDeploy(Promise<Void> completePromise, Deployment deployment) throws Exception {
+		onBeforeVerticleDeploy(deployment);
+		completePromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onBeforeVerticleDeploy(Promise, Deployment)}.
+	 */
+	public void onBeforeVerticleDeploy(Deployment deployment) throws Exception {
+	}
+
+	@Override
+	public void onAfterVerticleDeploy(Promise<Void> completePromise, Deployment deployment) throws Exception {
+		onAfterVerticleDeploy(deployment);
+		completePromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onAfterVerticleDeploy(Promise, Deployment)}.
+	 */
+	public void onAfterVerticleDeploy(Deployment deployment) throws Exception {
+	}
+
+	@Override
+	public void onBeforeVerticleUndeploy(Promise<Void> completePromise, Deployment deployment) throws Exception {
+		onBeforeVerticleUndeploy(deployment);
+		completePromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onBeforeVerticleUndeploy(Promise, Deployment)}.
+	 */
+	public void onBeforeVerticleUndeploy(Deployment deployment) throws Exception {
+	}
+
+	@Override
+	public void onAfterVerticleUndeploy(Promise<Void> completePromise, Deployment deployment) throws Exception {
+		onAfterVerticleUndeploy(deployment);
+		completePromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onAfterVerticleUndeploy(Promise, Deployment)}.
+	 */
+	public void onAfterVerticleUndeploy(Deployment deployment) throws Exception {
+	}
+
+	@Override
+	public void onVertxException(Promise<Void> completePromise, Throwable cause) throws Exception {
+		onVertxException(cause);
+		completePromise.complete();
+	}
+
+	/**
+	 * Synchronous version of {@link #onVertxException(Promise, Throwable)}.
+	 */
+	public void onVertxException(Throwable cause) throws Exception {
+	}
+}
```

Comment:
I don't know but I'd rather see the order reversed of public and private reversed. As we are not in C or C++ were we need forwared declaration, I think our silent convention should be to put protected and especially private fields and methods to the bottom, as usually users are only interested in public and maybe protected stuff... I would like to make their life easier and not need them scrolling 100 years 🚀

Reply:
Well, that's interesting. We've written the `TelestionVerticle` class in the exact same order :thinking: